test/core/end2end/end2end_tests.h

```diff
@@ -181,6 +181,36 @@ class CoreEnd2endTest : public ::testing::Test {
     void* p;
   };
 
+  // Safe notification to use for core e2e tests.
+  // Since when we're fuzzing we don't run background threads, the normal
+  // Notification type isn't safe to wait on (for some background timer to fire
+  // for instance...), consequently we need to use this.
+  class TestNotification {
```

Comment:
How is this related to the rest of this PR?  Did this change cause test failures without this, and if so, why?

Reply:
This was necessary to debug a failure that showed up (the tests will pass without this now, but next time they break it'll be hard to fix again).